mast_aladin_lite/table/mast_table.py

```diff
@@ -0,0 +1,172 @@
+
+        self.table = table
+
+        self.items = serialize(table)
+        mission = detect_mission(table)
+        columns = table.colnames
+        self.column_descriptions = get_column_descriptions(mission)
+
+        default_keys = [
+            'fileSetName',  # data products from Missions Mast
+            'source_id',    # Gaia
+            'MatchID',      # Hubble Source Catalog
+            'objID',        # PanSTARRS
+        ]
+        for default_key in default_keys:
+            if default_key in columns:
+                self.item_key = default_key
+                break
+
+        self.headers_avail = columns
+
+        # by default, remove the `s_region`` column
+        # from the visible columns in the widget:
+        if 's_region' in columns:
+            columns.remove('s_region')
+
+        self.headers_visible = columns
+
+        _table_widgets[len(_table_widgets)] = self
+
+    @observe('selected_rows')
+    def _on_row_selection(self, msg={}):
+        for func in self.row_select_callbacks:
+            func(msg)
+
+
+def get_current_table():
+    """
+    Return the last instantiated table widget.
+    """
+    latest_table_index = list(_table_widgets.keys())[-1]
+    return _table_widgets[latest_table_index]
+
+
+def update_mast_column_lists(update_column_descriptions=True, update_unique_columns=True):
+    """
+    Return a dictionary of columns in observation query
+    results from astroqery.mast.missions.MastMissions
+    that are unique to each mission.
+    """
+    from astroquery.mast.missions import MastMissions
+
+    mast = MastMissions()
+
+    columns_available = dict()
+    unique_columns = dict()
+    column_descriptions = defaultdict(list)
+
+    for mission in missions:
+        mast.mission = mission
+        column_list = mast.get_column_list()
+
+        for row in column_list.iterrows():
+            column_descriptions[mission].append(
+                {k: str(v).strip() for k, v in zip(column_list.colnames, row)}
+            )
+
+        column_names = column_list['name'].tolist()
+        columns_available[mission] = set(column_names)
+
+    if update_column_descriptions:
+        json_file = open(column_descriptions_path, 'w')
+        json.dump(column_descriptions, json_file)
+
+    for mission in missions:
+        mast.mission = mission
+        column_names = mast.get_column_list()['name']
+
+        other_missions = list(missions)
+        other_missions.remove(mission)
+        other_mission_columns = columns_available[other_missions[0]].union(
+            columns_available[other_missions[1]]
+        )
+        unique_columns[mission] = list(
+            columns_available[mission].difference(
+                other_mission_columns
+            )
+        )
+
+    if update_unique_columns:
+        json_file = open(unique_column_path, 'w')
+        json.dump(unique_columns, json_file, indent=4)
+
+    return unique_columns
+
+
+def detect_mission(table):
```

Comment:
question: are most missions, and the data returned from them, consistent enough that we can trust at least one of the unique columns to be present in a table always?

Reply:
suggestion: even if we are confident in this method, giving the user the ability to pass a mission id to the table might have some value.

Reply:
There aren't many missions supported in Missions Mast, but the columns are subject to change. We could just "hard code"/trust a specific column, but there's some chance that we need to change it later on.